### PR TITLE
Feature/flexible build

### LINKED ISSUE #86: Need to test building without a virtualenv
I just found out that we might not need maturin, we can just move and rename the file that was built by `cargo build` such as

On MacOS, rename `libstring_sum.dylib` to `string_sum.so`, on Windows `libstring_sum.dll` to `string_sum.pyd`, and on Linux `libstring_sum.so` to `string_sum.so`.

### PR DESCRIPTION
**Pull request by Ragarnoy**
Adds cargo make building

### CODE REVIEW

**Review by arlaine4 — APPROVED** [review]
I tried it, it seems to work just fine

**Review by arlaine4 — APPROVED** [review]

File: start_gomoku.sh [hunk 1]
```diff
@@ -36,29 +38,37 @@ PYTHON=src/PyInterface/gomoku.py
...
+if ! maturin develop --release; then
+    echo -e "\033[0;31mMaturin failed to build the project in a venv.\033[0m Will try to build without a virtual environment"
+    else
+      HAS_VENV=true
+    if ! pip3 install -r requirements.txt; then
+    echo -e "\033[0;31mCould not install requirements. Cannot build project\033[0m"
     exit 42
+    else
+      HAS_VENV=false
+    fi
```
Comment (demonoidv) [inline start_gomoku.sh]: Maybe an indent issue here (the first `else` should not be there)

**Review by demonoidv — APPROVED** [review]
LGTM! Awesome work, it's way easier to build the product now, thank you very much for that 👍 

I only noticed a little indentation issue in the shell script, you can fix it if you want or let it as is, both work for me.

File: start_gomoku.sh (hunk 1 above)
Comment (Ragarnoy) [reply start_gomoku.sh]: Fixed

Comment (demonoidv) [reply start_gomoku.sh]: That's a really nice fix 😂

**Comment by demonoidv** [comment]
Still good for me, feel free to merge whenever you want 👍

**Pull request merged**